libs/execution_context/ebpf_maps.c

```diff
@@ -557,11 +664,14 @@ _update_hash_map_entry(
     entry_count = ebpf_hash_table_key_count((ebpf_hash_table_t*)map->data);
 
     if ((entry_count == map->ebpf_map_definition.max_entries) &&
-        (ebpf_hash_table_find((ebpf_hash_table_t*)map->data, key, &value) != EBPF_SUCCESS))
+        (ebpf_hash_table_find((ebpf_hash_table_t*)map->data, key, &value) != EBPF_SUCCESS) &&
+        !_reap_oldest_map_entry(map))
         result = EBPF_INVALID_ARGUMENT;
     else
         result = ebpf_hash_table_update((ebpf_hash_table_t*)map->data, key, data, hash_table_operation);
 
+    _update_key_history(map, key, false);
```

Comment:
shouldn't this only be done in an EBPF_SUCCESS case?   In the 669 path, it's already done inside the ebpf_hash_table_find(), and in the 671 path if that fails there should be nothing to update.